### DISCUSSION TITLE
optimising-asp-net-core-apps-in-docker-avoiding-manually-copying-csproj-files-part-2/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#257] [Announcements]
# optimising-asp-net-core-apps-in-docker-avoiding-manually-copying-csproj-files-part-2/

In this post I show how you can optimise your ASP.NET Core Dockerfile for restore without having to list every project or create a tarball using scripts

https://andrewlock.net/optimising-asp-net-core-apps-in-docker-avoiding-manually-copying-csproj-files-part-2/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 25 2018, 05:23, in reply to [GaProgMan](https://disqus.com/by/gaprogman/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/257#discussioncomment-6045897)</em>

---

<p>Thanks, appreciate it! :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 06 2018, 01:27, in reply to [Yang Zhao](https://disqus.com/by/yangzhaodaniel/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/257#discussioncomment-6045904)</em>

---

<p>If they're standard, known, directories then you could tweak the scripts, but I don't think there's a generic approach. e.g.</p><p></p><pre><code><br>COPY src/Modules/*/*.csproj ./  <br>RUN for file in $(ls *.csproj); do mkdir -p src/Modules/${file%.*}/ &amp;&amp; mv $file src/Modules/${file%.*}/; done<br></code></pre>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 06 2020, 09:26, in reply to [cocowalla](https://disqus.com/by/cocowalla/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/257#discussioncomment-6045910)</em>

---

<p>Urgh, that sucks. I haven't started looking at BuildKit yet - if you figure out a solution before I do then let me know! :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 06 2020, 09:28, in reply to [Jindřich Cincibuch](https://disqus.com/by/jindichcincibuch/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/257#discussioncomment-6045912)</em>

---

<p>Maybe related to this? <a href="https://github.com/NuGet/Home/issues/8638" rel="nofollow noopener" title="https://github.com/NuGet/Home/issues/8638">https://github.com/NuGet/Home/issues/8638</a></p><p>I don't really know I'm afraid, not something I've seen</p>

[a comment by an automated account is not shown]

**Reply by angularsen** [reply]
This is the first viable working solution I've found for our rather complex monorepo with different folder structures and tons of shared code, thank you! It worked well for us.
